src/ovp_pipeline/commands/backfill_entities.py

```
@@ -85,7 +86,28 @@ def run(
     extraction_log = vault_dir / "60-Logs" / "entity-extractions.jsonl"
     extraction_log.parent.mkdir(parents=True, exist_ok=True)
 
+    # Dedup against prior runs: skip files already extracted unless --force.
+    # Without this, ``ovp-backfill-entities`` would re-process every file
+    # on every run (we hit this in the May 2026 history rerun: 274 files
+    # got 2-4 entries in the log, doubling token cost for no new data).
+    already_extracted: set[str] = set()
+    if not force and extraction_log.exists():
+        for line in extraction_log.read_text(encoding="utf-8").splitlines():
+            line = line.strip()
```

Ruff (0.15.12)

[warning] `for` loop variable `line` overwritten by assignment target

(PLW2901)